### PR TITLE
schema-filter-added-in-datalake-for-bucket

### LINKED ISSUE #7231: Treat buckets as schemas when ingesting with the Datalake connector
**Issue**
When ingesting with the Datalake connector, buckets are not treated as schemas, so applying the schema filters to the bucket's names is impossible.

### PR DESCRIPTION
**Pull request by codingwithabhi**
### Describe your changes :

I worked on the adding schema filter buckets in Datalake.
fix #7231 
#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] All new and existing tests passed.

#
### Reviewers
@open-metadata/ingestion

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/database/datalake.py [hunk 1]
```diff
@@ -164,9 +202,13 @@ def yield_database_schema(
...
-        paginator = self.client.get_paginator("list_objects_v2")
-        for page in paginator.paginate(**kwargs):
-            yield from page["Contents"]
+        try:
+            paginator = self.client.get_paginator("list_objects_v2")
+            for page in paginator.paginate(**kwargs):
+                yield from page["Contents"]
+        except Exception as exc:
+            logger.debug(traceback.format_exc())
+            logger.warning(f"Unexpected exception to yield s3 object [{page}]: {exc}")
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/source/database/datalake.py:211]: Just out of curiosity, do you know what error we are trying to catch?

**Review by TeddyCr — APPROVED** [review]
LGTM

**Comment by TeddyCr** [comment]
@codingwithabhi I think you'll need to push to your branch again.

File: ingestion/src/metadata/ingestion/source/database/datalake.py (hunk 1 above)
Comment (ulixius9) [reply ingestion/src/metadata/ingestion/source/database/datalake.py:211]: one possible error here is that we do not have enough access to list `list_objects_v2`

**Comment by TeddyCr** [comment]
@codingwithabhi could you fix code coverage by adding tests, please?

**Review by ulixius9 — APPROVED** [review]

**Pull request merged**